internal/broker/export_test.go

```diff
@@ -105,5 +108,26 @@ func (b *Broker) FetchUserInfo(sessionID string, cachedInfo *authCachedInfo) (st
 		return "", err
 	}
 
-	return b.userInfoFromClaims(uInfo, groups)
+	userInfo := b.userInfoFromClaims(uInfo, groups)
+	data, err := json.Marshal(userInfo)
+	if err != nil {
+		return "", err
+	}
+	var indented bytes.Buffer
+	err = json.Indent(&indented, data, "", "\t\t")
+	return indented.String(), err
+}
+
+var ErrTestChallengeDecodeError error
+
+func init() {
+	defaultDecodeRawChallenge := decodeRawChallenge
+	decodeRawChallenge = func(priv *rsa.PrivateKey, rawChallenge string) (decoded string, err error) {
```

Comment:
`decodeRawChallenge` is a function attached to the broker struct with a default, and ways to override it with `WithDecodeRawChallengeFunc()`.

That way, we keep parallelized tests and don’t change global state.